src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -89,6 +97,66 @@ public async Task<ExtractMethodResult> ExtractMethodAsync(CancellationToken canc
                 generatedCode.MethodNameAnnotation,
                 generatedCode.MethodDefinitionAnnotation,
                 cancellationToken).ConfigureAwait(false);
+
+            bool CanAddTo(Document document, SyntaxNode insertionPointNode, out OperationStatus status)
+            {
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxKinds = syntaxFacts.SyntaxKinds;
+                var codeGenService = document.GetLanguageService<ICodeGenerationService>();
+
+                if (insertionPointNode is null)
+                {
+                    status = OperationStatus.NoValidLocationToInsertMethodCall;
+                    return false;
+                }
+
+                var destination = insertionPointNode;
+                if (!LocalFunction)
+                {
+                    var mappedPoint = insertionPointNode.RawKind == syntaxKinds.GlobalStatement
+                        ? insertionPointNode.Parent
+                        : insertionPointNode;
+                    destination = mappedPoint.Parent ?? mappedPoint;
+                }
+
+                if (!codeGenService.CanAddTo(destination, document.Project.Solution, cancellationToken))
+                {
+                    status = OperationStatus.OverlapsHiddenPosition;
+                    return false;
+                }
+
+                status = OperationStatus.Succeeded;
+                return true;
+            }
+        }
+
+        private static async Task<(SemanticDocument analyzedDocument, InsertionPoint insertionPoint)> GetAnnotatedDocumentAndInsertionPointAsync(
+            SemanticDocument document,
+            AnalyzerResult analyzeResult,
+            SyntaxNode insertionPointNode,
+            CancellationToken cancellationToken)
```

Comment:
this is a move of the original code that annotated the variables of interest in the document.  but it has also been updated to annotate the insertion point as well.